src/chart/gauge/GaugeView.ts

```diff
@@ -674,7 +674,10 @@ class GaugeView extends ChartView {
                     seriesModel.get(['progress', 'show']) ? data.getItemVisual(idx, 'style').fill : autoColor
                 ) as string;
                 const labelEl = newDetailEls[idx];
-                const formatter = itemDetailModel.get('formatter');
+                let formatter = itemDetailModel.get('formatter');
+                if (isArray(formatter)) {
```

Comment:
Thank you for the contribution! In fact, besides setting the detail formatter for the whole series, it's also possible to customize the formatter for each gauge detail. Just set [`detail.formatter`](https://echarts.apache.org/option.html#series-gauge.data.detail.formatter) for each data item. Refer to this [demo](https://echarts.apache.org/examples/editor.html?c=gauge-multi-title&code=MYewdgzgLgBA5gQwK5wKYBEFQTAvDAbQCgYYBvE0mANwQBslUAuGAJgAYAaS0sBAW2YwA5AHEQIACbDuVGFACWUOkIpzSIAGaaIqKAGFUYKKgBOLAsIC0AFnYBSGSIAcD4QF0eMAL6yqkvQQFOhY1dS0dPUNjMwtrO0dOEQBOAFZHTzlvSl9KMNJaBiE7P14BIWEAIT0TUxkvRWVVLw1tXQMjWri3JOFXDK9cuQDsYNCWmAj26K7CYR6U9I9Sqk0QU34sWeEyQsZvAAlhQZy_fJp6RhYANi4vPkEWYQAFM01UYCh6uUaVcfVJm0op1YnMEk4-m5MlQhv5AmNyBMpsCYuYwQthGkMitSGsNltQTs9qhvIBgQmOWRyRHcAG4iCAAA6KcB4RGkXSmBSoCAWLznUhQACeDIqiBQqG-6gQYGAAAt1v8ATAIPKAO4sKCmRg4qgqkCqgCCACMQNQhJrtRN2QoAF5CACMzh1pCUqH4AGUhX82UrSKA6AqRABiABiBv0zlSzgpSuyANhcgZIAUqMVAIUoDATwZWFlTAA9PmALKsAB0yU47FLAHYADJlitV6vAe2lmyVzhl66cVusTul7td6j2-2pYBcVs2Ky9_vXKxd2d0LjsYDTtuV-cDtesTdzrsANQAzKXD_oq4ee-XL-2m_3GzWbZKAaqFJIoHmYE6rTAVGA4O-nn6J9wiBDpU0ILgXAGeMdQZUwQDgUxuR5H0AT1dV5C1VBnUmM1TDoBAGQ1LCcPgpAwEkfRCOIxgJgTKgEAADwUCBaxTZpfTIiiqKIzDaJgiZJCwBAWDFNBMGwHVfg4pU1mMd1bQdGw6J1EYghCVD1BfN8PxKb9ZVQBQ4FlKAWHtdtvzkqAFLtMyLN9f1A2EINtE0YC5CNBBgAAa0QkByMokAAzRYRkCgEB3KoE1TACUwACUEEkBQkBQi9LPWTYoG2XZLhJRw6JOUhPG8Ol2gASVRQoAApNHIz4FBZKqAEofTEjBhIIdh3FLYlWQAaiqotc1LUxpUkEB-GamAACoYHtdh2Ca0twpDBRGNQSQqtYJq6VINqJIQAh7W63r8AGob3xGsaJqm2b5sW5aQFW9bNu23b4GQcSOtYE7cv6wbhtGiibpau6FqWla1o2radsofhBX0WUEFMKBS3aAB5JkGrAKrzg5LkUOIAF-X8YTRM-9rsAKuRoW8WHfDYBbFppIA).

Reply:
oh～i get it ,thank you！I'm just trying to solve this [issue#18021](https://github.com/apache/echarts/issues/18021)🤣

Reply:
Anyway, thanks for your work! Issue #18021 raised by @na-am can be easily solved with the above approach. So I'm going to close this PR. Welcome your next contribution :) 😀